src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -263,7 +257,7 @@ class TaskReplaceActor(
   /**
     * @return whether [[instance]] has the new run spec version or an old one.
     */
-  def isOldInstance(instance: Instance): Boolean = instance.runSpecVersion != runSpec.version
+  def isOldInstance(instance: Instance): Boolean = instance.runSpecId == runSpec.id && instance.runSpecVersion != runSpec.version
```

Comment:
Not sure this check is really required since we do this above. What about throwing an exception instead since this is not a legal state
```suggestion
  def isOldInstance(instance: Instance): Boolean = {
    require(instance.runSpecId == runSpec.id)
    instance.runSpecVersion != runSpec.version
  }
```

Reply:
A classic case of paranoid programming? Whatever...